api/lib/domain/services/certification-challenges-service.js

```diff
@@ -8,9 +8,21 @@ const KnowledgeElement = require('../models/KnowledgeElement');
 const UserCompetence = require('../models/UserCompetence');
 const Challenge = require('../models/Challenge');
 const challengeRepository = require('../../infrastructure/repositories/challenge-repository');
+const skillRepository = require('../../infrastructure/repositories/skill-repository');
 const answerRepository = require('../../infrastructure/repositories/answer-repository');
 const knowledgeElementRepository = require('../../infrastructure/repositories/knowledge-element-repository');
 
+function _challengeSnapshotsFromAnswersAndKnowledgeElements(answers, knowledgeElements) {
+  return answers.map((answer) => {
+    const associatedKnowledgeElements = knowledgeElements.filter((knowledgeElement) => knowledgeElement.answerId === answer.id);
+    return {
+      challengeId: answer.challengeId,
+      competenceId: associatedKnowledgeElements[0].competenceId, // Ici on doit être confiant dans le fait qu'une answer lié à plusieurs KE (QROCmDep) soit liés à des skills de la même compétence
```

Comment:
C'est plutôt le challenge qui est lié à des skills non ?
```suggestion
      competenceId: associatedKnowledgeElements[0].competenceId, // Ici on doit être confiant dans le fait qu'un challenge lié à plusieurs skills (QROCmDep) soit liée à des skills de la même compétence
```

Reply:
Oui, c'est un raccourci, peut-être que ça c'est mieux :
```suggestion
      // Ici on doit être confiant dans le fait qu'une réponse liée à
      // plusieurs KEs (QROCmDep) soit liée à une épreuve liée à des acquis
      // de la même compétence
      competenceId: associatedKnowledgeElements[0].competenceId,
```

Reply:
Le truc qui m'embêtait au départ c'est qu'une réponse liée à plusieurs KE, c'est la norme à cause de l'inférence ! Bien sûr les KE qui arrivent ici ne sont que les _directs_, mais sans avoir le contexte c'est pas évident de s'en souvenir.